### PR TITLE
FEM-2505 FB ADS plugin first drop

### PR DESCRIPTION
**Pull request by giladna**

### CODE REVIEW

File: fbadsplugin/src/main/java/com/kaltura/playkit/plugins/fbads/fbinstream/FBInstreamPlugin.java [hunk 1]
```diff
@@ -360,6 +399,98 @@ public void onLoggingImpression(Ad ad) {
...
+        if (adResponse != null &&
+                adResponse.getCapabilities() != null &&
+                adResponse.getCapabilities().getSkipButton() != null &&
+                adResponse.getCapabilities().getSkipButton().getSkippableSeconds() != null) {
+            log.d("FB Instream AdResponse skip = " + adResponse.getCapabilities().getSkipButton().getSkippableSeconds());
+            adInfo.setAdSkipOffset(Long.valueOf(adResponse.getCapabilities().getSkipButton().getSkippableSeconds()));
+        }
+        adInfo.setAdHeight(ad.getMeasuredHeight());
+        adInfo.setAdWidth(ad.getMeasuredWidth());
+    }
```
Comment (noamtamim) [inline fbadsplugin/src/main/java/com/kaltura/playkit/plugins/fbads/fbinstream/FBInstreamPlugin.java]: Put everything in try-catch.

File: fbadsplugin/src/main/java/com/kaltura/playkit/plugins/fbads/fbinstream/FBInStreamAdBreak.java [hunk 2]
```diff
@@ -13,6 +13,14 @@ public FBInStreamAdBreak(AdPositionType adBreakType, long adBreakTime, List<FBIn
         this.adBreakType = adBreakType;
         this.adBreakTime = adBreakTime;
         this.fbInStreamAdList = fbInStreamAdList;
+        if (fbInStreamAdList != null && fbInStreamAdList.size() > 0) {
+            for (int indx = 0 ; indx < fbInStreamAdList.size() ; indx++) {
+                if (fbInStreamAdList.get(indx) != null) {
```
Comment (noamtamim) [inline fbadsplugin/src/main/java/com/kaltura/playkit/plugins/fbads/fbinstream/FBInStreamAdBreak.java]: Extract variable `fbInStreamAdList.get(indx)`.

File: fbadsplugin/src/main/java/com/kaltura/playkit/plugins/fbads/fbinstream/FBInstreamPlugin.java [hunk 3]
```diff
@@ -380,14 +538,15 @@ private int pxToDP(int px) {
...
-        log.d("Start onUpdateConfig");
+        log.d("FB Instream Ads onUpdateConfig");
         adConfig = parseConfig(config);
         buildAdBreaksMap();
     }
 
     @Override
     protected void onApplicationPaused() {
         log.d("Start onApplicationPaused");
+        //timer.cancel();
```
Comment (noamtamim) [inline fbadsplugin/src/main/java/com/kaltura/playkit/plugins/fbads/fbinstream/FBInstreamPlugin.java]: Remove

File: fbadsplugin/src/main/java/com/kaltura/playkit/plugins/fbads/fbinstream/FBInstreamPlugin.java [hunk 4]
```diff
@@ -406,22 +565,26 @@ protected void onDestroy() {
             adContainer.removeAllViews();
             adContainer = null;
         }
+        //cancelTimer();
```
Comment (noamtamim) [inline fbadsplugin/src/main/java/com/kaltura/playkit/plugins/fbads/fbinstream/FBInstreamPlugin.java]: Remove

File: fbadsplugin/src/main/java/com/kaltura/playkit/plugins/fbads/fbinstream/FBInstreamPlugin.java [hunk 5]
```diff
@@ -430,6 +593,7 @@ public void destroyAdsManager() {
...
```
Comment (noamtamim) [inline fbadsplugin/src/main/java/com/kaltura/playkit/plugins/fbads/fbinstream/FBInstreamPlugin.java]: Remove

**Review by noamtamim — CHANGES_REQUESTED** [review]

File: fbadsplugin/src/main/java/com/kaltura/playkit/plugins/fbads/fbinstream/FBInStreamAdBreak.java (hunk 2 above)
Comment (giladna) [reply fbadsplugin/src/main/java/com/kaltura/playkit/plugins/fbads/fbinstream/FBInStreamAdBreak.java]: done

**Review by noamtamim — APPROVED** [review]

**Pull request merged**